container/container.go

```diff
@@ -667,15 +667,31 @@ func (container *Container) BuildCreateEndpointOptions(n libnetwork.Network, epC
 
 	if epConfig != nil {
 		ipam := epConfig.IPAMConfig
-		if ipam != nil && (ipam.IPv4Address != "" || ipam.IPv6Address != "" || len(ipam.LinkLocalIPs) > 0) {
+
+		if ipam != nil {
 			var ipList []net.IP
+			var ip, ip6 net.IP
+
 			for _, ips := range ipam.LinkLocalIPs {
-				if ip := net.ParseIP(ips); ip != nil {
-					ipList = append(ipList, ip)
+				if linkip := net.ParseIP(ips); linkip == nil && ips != "" {
```

Comment:
if you also add a `llIP` to the above var block, you can improve readability here dropping the else block

```
    if llIP  = net.ParseIP(ips); linkip == nil && ips != "" {
        return nil, fmt.Errorf("Error....)
    }
    ipList = append(ipList, llIP)
}